airbyte-lib/airbyte_lib/telemetry.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+import datetime
+import os
+from typing import Any, Dict, Optional
+
+import requests
+from airbyte_lib.version import get_version
+
+# TODO: Use production tracking key
+TRACKING_KEY = "jxT1qP9WEKwR3vtKMwP9qKhfQEGFtIM1" or str(os.environ.get("AIRBYTE_TRACKING_KEY"))
+
+
+def track(source: Dict[str, str], target: str, state: str, number_of_records: Optional[int] = None):
+    # If DO_NOT_TRACK is set, we don't send any telemetry
+    if os.environ.get("DO_NOT_TRACK"):
+        return
+
+    current_time = datetime.datetime.utcnow().isoformat()
+    payload: Dict[str, Any] = {
+        "anonymousId": "airbyte-lib-user",
+        "event": "sync",
+        "properties": {
+            "version": get_version(),
+            "source": source,
+            "state": state,
+            "target": target,
+            # explicitly set to 0.0.0.0 to avoid leaking IP addresses
+            "ip": "0.0.0.0",
+        },
+        "timestamp": current_time,
+    }
+    if number_of_records is not None:
+        payload["properties"]["number_of_records"] = number_of_records
+
+    # Do not handle the response, we don't want to block the execution
+    requests.post("https://api.segment.io/v1/track", auth=(TRACKING_KEY, ""), json=payload)
```

Comment:
Totally optional, but a common convention in Python if you want to call out that the intentional dropping of a return value - you can use "_ = ..."

This isn't necessary by any means, but will inform the reader "yes, I know there's a return value and no, I don't want to use it." 😄 

```suggestion
    # Do not handle the response, we don't want to block the execution
    _ = requests.post("https://api.segment.io/v1/track", auth=(TRACKING_KEY, ""), json=payload)
```